### PR TITLE
fix: store favorites regardless of languages

### LINKED ISSUE #1110: Favorites are not displayed
### Describe the bug

No favorites are displayed in the side menu in the German language, but it works perfectly in all other selectable languages.

### What happened?

A bug happened!

### System information

On Synology 1621xs+ as Docker/Container Version.

### Where did you encounter the bug?

Other (installations, docker, etc.)

### PR DESCRIPTION
**Pull request by sharevb**
Fix #1110

### CODE REVIEW

File: src/tools/tools.store.ts [hunk 1]
```diff
@@ -43,15 +45,16 @@ export const useToolStore = defineStore('tools', () => {
     newTools: computed(() => tools.value.filter(({ isNew }) => isNew)),
 
     addToolToFavorites({ tool }: { tool: MaybeRef<Tool> }) {
-      favoriteToolsName.value.push(get(tool).name);
+      favoriteToolsName.value.push(get(tool).path);
     },
 
     removeToolFromFavorites({ tool }: { tool: MaybeRef<Tool> }) {
-      favoriteToolsName.value = favoriteToolsName.value.filter(name => get(tool).name !== name);
+      favoriteToolsName.value = favoriteToolsName.value.filter(name => get(tool).name !== name && get(tool).path !== name);
```
Comment (CorentinTh) [inline src/tools/tools.store.ts:52]:
```suggestion
      favoriteToolsName.value = favoriteToolsName.value.filter(name => get(tool).path !== name);
```

File: src/tools/tools.store.ts [hunk 2]
```diff
@@ -43,15 +45,16 @@ export const useToolStore = defineStore('tools', () => {
...
     },
 
     isToolFavorite({ tool }: { tool: MaybeRef<Tool> }) {
-      return favoriteToolsName.value.includes(get(tool).name);
+      return favoriteToolsName.value.includes(get(tool).name)
+        || favoriteToolsName.value.includes(get(tool).path);
```
Comment (CorentinTh) [inline src/tools/tools.store.ts:57]:
```suggestion
      return favoriteToolsName.value.includes(get(tool).path);
```

File: src/tools/tools.store.ts [hunk 3]
```diff
@@ -23,16 +24,17 @@ export const useToolStore = defineStore('tools', () => {
...
+        path,
         components,
       }))
       .value();
   });
 
   const favoriteTools = computed(() => {
     return favoriteToolsName.value
-      .map(favoriteName => tools.value.find(({ name }) => name === favoriteName))
+      .map(favoriteName => tools.value.find(({ name, path }) => name === favoriteName || path === favoriteName))
```
Comment (CorentinTh) [inline src/tools/tools.store.ts:37]:
```suggestion
      .map(favoritePath => tools.value.find(({ path }) => path === favoritePath))
```

File: src/tools/tools.store.ts [hunk 4]
```diff
@@ -23,16 +24,17 @@ export const useToolStore = defineStore('tools', () => {
...
+      .map((components, name, path) => ({
         name,
+        path,
         components,
       }))
       .value();
   });
 
   const favoriteTools = computed(() => {
     return favoriteToolsName.value
```
Comment (CorentinTh) [inline src/tools/tools.store.ts:36]:
```suggestion
    return favoriteToolsPaths.value
```

**Review by CorentinTh — CHANGES_REQUESTED** [review]
Maybe no need for continuing to use the `name` along the path ?

**Comment by sharevb** [comment]
> Maybe no need for continuing to use the `name` along the path ?

I used both in order that current favorites are not loss by this change. Else, I think all favorites will be lost for everyone, no ?

**Review by CorentinTh — APPROVED** [review]

**Pull request merged**